### PR TITLE
Add mermaid charts support to `swift package describe`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
This adds support for new `--type mermaid` option on `swift package describe`.

This format is supported by the GitHub Markdown renderer and some Markdown preview extensions in IDEs.

I'm using different shapes to denote difference between products and targets. `| Foo |` is a target, `|| Bar ||` is a product, and hexagonal shapes declared with `{{ Baz }}` are products from package dependencies. I initially tried to include a legend subgraph, but that makes layout quite ugly even for simple packages, so I decided to exclude that by default. The codepath for rendering the legend is still there, so we can reenable it at any point or make conditional on some new CLI flag if needed.

Here's example output for the swift-syntax package:

```mermaid
flowchart TB
    subgraph swift-syntax
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftSyntaxTestSupportTest(SwiftSyntaxTestSupportTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftSyntaxTest(SwiftSyntaxTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftSyntaxMacrosTestSupportTests(SwiftSyntaxMacrosTestSupportTests)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftSyntaxBuilderTest(SwiftSyntaxBuilderTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftRefactorTest(SwiftRefactorTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftParserTest(SwiftParserTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftParserDiagnosticsTest(SwiftParserDiagnosticsTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftOperatorsTest(SwiftOperatorsTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftIDEUtilsTest(SwiftIDEUtilsTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftDiagnosticsTest(SwiftDiagnosticsTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftCompilerPluginTest(SwiftCompilerPluginTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:SwiftBasicFormatTest(SwiftBasicFormatTest)
        product:swift-syntaxPackageTests[[swift-syntaxPackageTests]]-->target:PerformanceTest(PerformanceTest)
        product:SwiftBasicFormat[[SwiftBasicFormat]]-->target:SwiftBasicFormat(SwiftBasicFormat)
        product:SwiftCompilerPlugin[[SwiftCompilerPlugin]]-->target:SwiftCompilerPlugin(SwiftCompilerPlugin)
        product:SwiftCompilerPluginMessageHandling[[SwiftCompilerPluginMessageHandling]]-->target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)
        product:SwiftDiagnostics[[SwiftDiagnostics]]-->target:SwiftDiagnostics(SwiftDiagnostics)
        product:SwiftIDEUtils[[SwiftIDEUtils]]-->target:SwiftIDEUtils(SwiftIDEUtils)
        product:SwiftOperators[[SwiftOperators]]-->target:SwiftOperators(SwiftOperators)
        product:SwiftParser[[SwiftParser]]-->target:SwiftParser(SwiftParser)
        product:SwiftParserDiagnostics[[SwiftParserDiagnostics]]-->target:SwiftParserDiagnostics(SwiftParserDiagnostics)
        product:SwiftRefactor[[SwiftRefactor]]-->target:SwiftRefactor(SwiftRefactor)
        product:SwiftSyntax[[SwiftSyntax]]-->target:SwiftSyntax(SwiftSyntax)
        product:SwiftSyntaxBuilder[[SwiftSyntaxBuilder]]-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        product:SwiftSyntaxMacros[[SwiftSyntaxMacros]]-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        product:SwiftSyntaxMacroExpansion[[SwiftSyntaxMacroExpansion]]-->target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)
        product:SwiftSyntaxMacrosTestSupport[[SwiftSyntaxMacrosTestSupport]]-->target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)
        target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)-->target:SwiftBasicFormat(SwiftBasicFormat)
        target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)-->target:SwiftSyntax(SwiftSyntax)
        target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)-->target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)
        target:SwiftSyntaxTestSupportTest(SwiftSyntaxTestSupportTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftSyntaxTestSupportTest(SwiftSyntaxTestSupportTest)-->target:SwiftParser(SwiftParser)
        target:SwiftSyntaxTest(SwiftSyntaxTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftSyntaxTest(SwiftSyntaxTest)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntaxTest(SwiftSyntaxTest)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftSyntaxMacrosTestSupportTests(SwiftSyntaxMacrosTestSupportTests)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntaxMacrosTestSupportTests(SwiftSyntaxMacrosTestSupportTests)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntaxMacrosTestSupportTests(SwiftSyntaxMacrosTestSupportTests)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftSyntaxMacrosTestSupportTests(SwiftSyntaxMacrosTestSupportTests)-->target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)
        target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)-->target:SwiftParser(SwiftParser)
        target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)-->target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)
        target:SwiftSyntaxMacros(SwiftSyntaxMacros)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntaxMacros(SwiftSyntaxMacros)-->target:SwiftParser(SwiftParser)
        target:SwiftSyntaxMacros(SwiftSyntaxMacros)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntaxMacros(SwiftSyntaxMacros)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftOperators(SwiftOperators)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftParser(SwiftParser)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)
        target:SwiftSyntaxMacroExpansionTest(SwiftSyntaxMacroExpansionTest)-->target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)
        target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)-->target:SwiftOperators(SwiftOperators)
        target:SwiftSyntaxBuilderTest(SwiftSyntaxBuilderTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftSyntaxBuilderTest(SwiftSyntaxBuilderTest)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)-->target:SwiftBasicFormat(SwiftBasicFormat)
        target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)-->target:SwiftParser(SwiftParser)
        target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)-->target:SwiftParserDiagnostics(SwiftParserDiagnostics)
        target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:_InstructionCounter(_InstructionCounter)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftBasicFormat(SwiftBasicFormat)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftCompilerPlugin(SwiftCompilerPlugin)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftIDEUtils(SwiftIDEUtils)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntax509(SwiftSyntax509)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntax510(SwiftSyntax510)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntax511(SwiftSyntax511)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftSyntaxMacrosTestSupport(SwiftSyntaxMacrosTestSupport)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftParser(SwiftParser)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftParserDiagnostics(SwiftParserDiagnostics)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftOperators(SwiftOperators)
        target:SwiftSyntax-all(SwiftSyntax-all)-->target:SwiftRefactor(SwiftRefactor)
        target:SwiftSyntax(SwiftSyntax)-->target:SwiftSyntax509(SwiftSyntax509)
        target:SwiftSyntax(SwiftSyntax)-->target:SwiftSyntax510(SwiftSyntax510)
        target:SwiftSyntax(SwiftSyntax)-->target:SwiftSyntax511(SwiftSyntax511)
        target:SwiftRefactorTest(SwiftRefactorTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftRefactorTest(SwiftRefactorTest)-->target:SwiftRefactor(SwiftRefactor)
        target:SwiftRefactor(SwiftRefactor)-->target:SwiftBasicFormat(SwiftBasicFormat)
        target:SwiftRefactor(SwiftRefactor)-->target:SwiftParser(SwiftParser)
        target:SwiftRefactor(SwiftRefactor)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftRefactor(SwiftRefactor)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftParserTest(SwiftParserTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftParserTest(SwiftParserTest)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftParserTest(SwiftParserTest)-->target:SwiftOperators(SwiftOperators)
        target:SwiftParserTest(SwiftParserTest)-->target:SwiftParser(SwiftParser)
        target:SwiftParserTest(SwiftParserTest)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftParserDiagnosticsTest(SwiftParserDiagnosticsTest)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftParserDiagnosticsTest(SwiftParserDiagnosticsTest)-->target:SwiftParserDiagnostics(SwiftParserDiagnostics)
        target:SwiftParserDiagnostics(SwiftParserDiagnostics)-->target:SwiftBasicFormat(SwiftBasicFormat)
        target:SwiftParserDiagnostics(SwiftParserDiagnostics)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftParserDiagnostics(SwiftParserDiagnostics)-->target:SwiftParser(SwiftParser)
        target:SwiftParserDiagnostics(SwiftParserDiagnostics)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftParser(SwiftParser)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftOperatorsTest(SwiftOperatorsTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftOperatorsTest(SwiftOperatorsTest)-->target:SwiftOperators(SwiftOperators)
        target:SwiftOperatorsTest(SwiftOperatorsTest)-->target:SwiftParser(SwiftParser)
        target:SwiftOperators(SwiftOperators)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftOperators(SwiftOperators)-->target:SwiftParser(SwiftParser)
        target:SwiftOperators(SwiftOperators)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftIDEUtilsTest(SwiftIDEUtilsTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftIDEUtilsTest(SwiftIDEUtilsTest)-->target:SwiftIDEUtils(SwiftIDEUtils)
        target:SwiftIDEUtilsTest(SwiftIDEUtilsTest)-->target:SwiftParser(SwiftParser)
        target:SwiftIDEUtilsTest(SwiftIDEUtilsTest)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftIDEUtils(SwiftIDEUtils)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftIDEUtils(SwiftIDEUtils)-->target:SwiftParser(SwiftParser)
        target:SwiftDiagnosticsTest(SwiftDiagnosticsTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftDiagnosticsTest(SwiftDiagnosticsTest)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftDiagnosticsTest(SwiftDiagnosticsTest)-->target:SwiftParser(SwiftParser)
        target:SwiftDiagnosticsTest(SwiftDiagnosticsTest)-->target:SwiftParserDiagnostics(SwiftParserDiagnostics)
        target:SwiftDiagnostics(SwiftDiagnostics)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftCompilerPluginTest(SwiftCompilerPluginTest)-->target:SwiftCompilerPlugin(SwiftCompilerPlugin)
        target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)-->target:SwiftDiagnostics(SwiftDiagnostics)
        target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)-->target:SwiftOperators(SwiftOperators)
        target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)-->target:SwiftParser(SwiftParser)
        target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)-->target:SwiftSyntax(SwiftSyntax)
        target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)-->target:SwiftSyntaxMacroExpansion(SwiftSyntaxMacroExpansion)
        target:SwiftCompilerPlugin(SwiftCompilerPlugin)-->target:SwiftCompilerPluginMessageHandling(SwiftCompilerPluginMessageHandling)
        target:SwiftCompilerPlugin(SwiftCompilerPlugin)-->target:SwiftSyntaxMacros(SwiftSyntaxMacros)
        target:SwiftBasicFormatTest(SwiftBasicFormatTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:SwiftBasicFormatTest(SwiftBasicFormatTest)-->target:SwiftBasicFormat(SwiftBasicFormat)
        target:SwiftBasicFormatTest(SwiftBasicFormatTest)-->target:SwiftSyntaxBuilder(SwiftSyntaxBuilder)
        target:SwiftBasicFormat(SwiftBasicFormat)-->target:SwiftSyntax(SwiftSyntax)
        target:PerformanceTest(PerformanceTest)-->target:_InstructionCounter(_InstructionCounter)
        target:PerformanceTest(PerformanceTest)-->target:_SwiftSyntaxTestSupport(_SwiftSyntaxTestSupport)
        target:PerformanceTest(PerformanceTest)-->target:SwiftIDEUtils(SwiftIDEUtils)
        target:PerformanceTest(PerformanceTest)-->target:SwiftParser(SwiftParser)
        target:PerformanceTest(PerformanceTest)-->target:SwiftSyntax(SwiftSyntax)
    end
```

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

File: Sources/Commands/PackageTools/Describe.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ extension SwiftPackageTool {
         @OptionGroup(visibility: .hidden)
         var globalOptions: GlobalOptions
         
-        @Option(help: "json | text")
+        @Option(help: "json | text | mermaid")
```
Comment (rauhul) [inline Sources/Commands/PackageTools/Describe.swift:29]: Could this be "output format"? Newer versions of argument parser should handle printing the valid options.

Comment (MaxDesiatov) [reply Sources/Commands/PackageTools/Describe.swift:29]: Yes, I can do that, but we have to update to a new version of ArgumentParser first, and that's a quite complicated process that involves a coordinated set of PRs across all of toolchain repositories that depend on it. Seemed out of scope for this specific PR.

Comment (rauhul) [reply Sources/Commands/PackageTools/Describe.swift:29]: I did not know that! We can make an issue for modernizing the argument helps whenever the arg parser is bumped next.

Comment (MaxDesiatov) [reply Sources/Commands/PackageTools/Describe.swift:29]: https://github.com/apple/swift-package-manager/issues/7290

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

File: Sources/Commands/Utilities/DescribedPackage.swift [hunk 2]
```diff
@@ -258,232 +258,3 @@ struct DescribedPackage: Encodable {
         }
     }
 }
-
-
-public struct PlainTextEncoder {
```
Comment (MaxDesiatov) [inline Sources/Commands/Utilities/DescribedPackage.swift:263]: Moved to a separate file.

**Review by bnbarham — APPROVED** [review]
What's the `|| Foo ||` vs `| Foo |`?

**Comment by MaxDesiatov** [comment]
> What's the `|| Foo ||` vs `| Foo |`?

I'm using different shapes to denote difference between products and targets. | Foo | is a target, || Bar || is a product, and hexagonal shapes declared with {{ Baz }} are products from package dependencies. I initially tried to include a legend subgraph by default, but that makes layout quite ugly even for simple packages, so I decided to exclude that by default. The codepath for rendering the legend is still there, so we can reenable it at any point if needed, or make conditional on some new CLI flag.

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**